pkg/container-collection/container-collection.go

```diff
@@ -155,6 +147,47 @@ func (cc *ContainerCollection) RemoveContainer(id string) {
 	// the notification handler, and they expect the container to still be
 	// present.
 	cc.containers.Delete(id)
+
+	// Make this operation atomic, as RemoveContainer() could be called concurrently, which could result in
+	// dirty map contents
+	cc.mu.Lock()
+	defer cc.mu.Unlock()
```

Comment:
I wanted have all operations regarding the maps in `AddContainer()` and `RemoveContainer()` basically as one transaction - otherwise when calling both functions concurrently there's no guarantee for consistent behavior.

See for example the `containersByNetNs`: when adding a container, a lookup by netNs is happening (which by itself would be thread safe), but the resulting array is being modified (the new container added) and then stored back to the map. That must not happen concurrently.